crates/starknet/src/state.rs

```diff
@@ -11,9 +13,32 @@ use starknet_types::DevnetResult;
 
 use crate::traits::{StateChanger, StateExtractor};
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub(crate) struct StarknetState {
-    state: InMemoryStateReader,
+    pub state: InMemoryStateReader,
+    pub pending_state: CachedState<InMemoryStateReader>,
+}
+
+impl StarknetState {
+    // this is used to copy "persistent" data that is present in "state" variable into "pending_state"
+    // this is done, because "pending_state" doesnt hold a reference to state, but rather a copy.
+    pub(crate) fn equalize_states(&mut self) {
```

Comment:
I need an explanation about this. Is it better to call this `synchronize_states` or is my understanding wrong?

Reply:
Method renamed in [78293ff](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/20/commits/78293ff8b3552dfc6578ad274ae11f41f60ad4d8)